ga docs for Kommander 1.2 (#3326)
mesosphere/dcos-docs-site: pages/ksphere/kommander/1.2/release-notes/index.md

[button color="purple" href="https://support.d2iq.com/s/entitlement-based-product-downloads"]Download Konvoy[/button]

To get started with Kommander, [download](https://docs.d2iq.com/ksphere/konvoy/latest/download/) and [install](https://docs.d2iq.com/ksphere/konvoy/latest/install/) the latest version of Konvoy.

**NOTE:** You must be a registered user and logged on to the support portal to download this product. New customers must contact your sales representative or <a href="mailto:sales@d2iq.com">sales@d2iq.com</a> before attempting to download or install Konvoy.</p>

# Release Summary 
Kommander provides a command center for all your cloud native management needs in public Information as a Service (IaaS), on-premises, and edge environments. Kommander provides a multi-tenant experience to create, secure, and configure Kubernetes clusters and cloud native workloads. Additionally, Kommander enables teams to unlock federated cost management across multiple clusters, whether they are a new Konvoy cluster or an existing 3rd party/DIY distribution installation.

# Supported Versions
| Kubernetes Support | Version |
| ------------------ | ------- |

> | **Minimum**        | 1.15.0  |

Reviewer: ```suggestion
| **Minimum**        | 1.16.0  |
```

Reviewer: @juliangieseke should we update this? (1.2 branch) https://github.com/mesosphere/kubeaddons-kommander/blob/1.2.x/addons/kommander/1.2/kommander.yaml#L28

Reviewer 2: Im not even sure we still need that. IIRC it comes from times before we had the utility server 🙈

Reviewer: probably safer to leave it and just state 1.16 to avoid confusion and then take it out after GA if we really don't need it

Reviewer 2: 1.2.x branch is updated with 1.16.0 👍

> | **Maximum**        | 1.17.x  |

Reviewer: ```suggestion
| **Maximum**        | 1.18.x  |
```

> | **Default**        | 1.17.3  |

Reviewer: Basing it off this: https://github.com/mesosphere/yakcl/blob/v0.6.5/Makefile#L41

```suggestion
| **Default**        | 1.18.8  |
```

# New Features and Capabilities 

## Air gapped environments
Kommander now supports installing and running in an air gapped environment, either on premises or in the cloud. Air gapping is ideal for instances where high-security is a must or where Internet connectivity is either undesirable or unavailable. For more details on setting up Kommander in an air gapped environment, see [Install Kommander air gapped][install_airgapped].

## Component versions

> - Addon: 1.2.0-26

Reviewer 3: need to update this with final addon version

> - Chart: 0.12.3

Reviewer 3: same, need to update with final chart version

- kommander-federation (yakcl): 0.6.2

> - kommander-licensing (yakcl): 0.6.2

Reviewer: @drewvanstone @juliangieseke pls have someone review and verify what versions will be used for components. yakcl 0.6.5 is latest but i suspect we will have 1 more release to include for the GA

Reviewer 2: yeah, I will update this once we actually tagged the release.

- UI: 6.37.0

> - kommander-karma: 0.3.10

Reviewer 3: for these parent charts (kommander-karma, kommander-thanos, kubecost) do we want to also/instead include the actual component versions (karma, thanos, kubecost) which is probably the more useful information?

Reviewer: good question, i vote for actual version (appversion?) but i leave it to yall to decide

Reviewer 2: hm, main idea behind these component was to guide the user (us) to the next thing to check. if we'd have `karma 1.2.3` here, we'd not know which chart version got included and it makes it harder to debug IMHO.

that said, maybe having both would be helpful? not sure how deep we should go tho, I feel like having dep->dep->dep is kinda useless at some point.

Author: I also prefer the most current version of all components, not just those that have been updated. We try to do this with DCOS as well, and wnat to do that for Konvoy. Consistency would be best.

Author: ooh, I like the idea of having both. list all of them and figure out a way to format the updated ones differently?

Author: Could we do this for Konvoy too? Or is it too late?

Reviewer 4: Couldn't we do it for Konvoy? I feel like it's not too late. Can always add to it/improve the docs.

Reviewer 3: App versions in case we want to include them:
```
- karma: 0.70
- thanos: 0.10.1
- cost-analyzer: 1.69.0
- grafana: 6.6.0
```

- kubeaddons-catalog: 0.1.15
- kommander-thanos: 0.1.15
- kubecost: 0.1.15
- grafana: 4.6.3

> ## Fixed and Improved Issues 

Reviewer: Is this everything? do we want to break these up into component-related updates? or maybe since the list is only 10 bullets long it's probably fine to leave as is.

@drewvanstone can you review? I think your team addressed more improvements than is listed here.

Reviewer 3: Maybe missing:
```
- Updated cert-manager resources to v1
- Added automatic creation of KommanderProjectRole and VirtualGroupKommanderProjectRoleBinding objects based on Workspace objects
- Fixed access to kubecost UI for some cluster roles
- Created the serviceaccount needed for deletion jobs on delete (#220)
- Added --ignore-not-found to resource cleanup jobs to ignore errors if resource has already been deleted
- Fixed metric collection for kubefed, kommander-federation, and kommander-licensing
```

- Improved Access Control for editing/deleting actions. 
- Changed routing from hash-based to history-based to allow redirects after login.
- Added SAML IDP.
- Moved provisioning code to Konvoy, Kommander now uses Konvoy to provision clusters.
- Added a setting to configure thresholds for resource warnings. 
- Added the ability to show metrics in attached EKS clusters.
- Added Infrastructure provider management on global level.
- Fixed an issue where logout was not deleting any cookies.
- Fixed UI crashes when kubecost was down.

> - Fixed catalog API occasionaly returned 500.

Reviewer 5: ```suggestion
- Fixed an issue where the catalog API occasionally returned 500.
```

> - Fixed deleting cluster labels was impossible.

Reviewer 5: ```suggestion
- Fixed an issue that prevented deleting cluster labels.
```

Reviewer 4: comitting this - makes sense to me

> - Fixed counting self-attached host cluster against license count.

Reviewer 5: This is very unclear, and probably important.  Does the following capture the issue?

```suggestion
- Fixed an issue that erroneously counted the self-attached host cluster against the license count.
```

Reviewer 4: @cmastr @russurquhart1 you good with that? ☝️

> - Smaller UX Bugs and Improvements.

Reviewer 5: I am not sure the 'size' of the bug is relevant here.   How about this:
```suggestion
- Numerous UX Bugfixes and Improvements.
```

Reviewer 4: I dig it. Thanks @cneth

> - Numerous UX Bugfixes and Improvements.

Reviewer 4: ```suggestion
- Numerous UX Bug fixes and Improvements.
```

<!-- markdownlint-disable MD034 -->

# Release notes for Kommander 1.2

> Kommander 1.2 was released on 11, November 2020 

Reviewer 4: This still valid?

Reviewer 3: I just cut GA tags so.. maybe `16, November 2020` instead?

# Release Summary 
Kommander provides a command center for all your cloud native management needs in public Information as a Service (IaaS), on-premises, and edge environments. Kommander provides a multi-tenant experience to create, secure, and configure Kubernetes clusters and cloud native workloads. Additionally, Kommander enables teams to unlock federated cost management across multiple clusters, whether they are a new Konvoy cluster or an existing 3rd party/DIY distribution installation.

# Supported Versions
| Kubernetes Support | Version |
| ------------------ | ------- |
| **Minimum**        | 1.16.0  |
| **Maximum**        | 1.18.x  |
| **Default**        | 1.18.8  |

# New Features and Capabilities 

## Air gapped environments
Kommander now supports installing and running in an air gapped environment, either on premises or in the cloud. Air gapped environments are ideal when high-security is a must or when Internet connectivity is undesirable or unavailable. For information on setting up Kommander in an air gapped environment, see [Install Kommander air gapped][install_airgapped].

## Component versions

> - Addon: 1.2.0-26

Reviewer 3: ```suggestion
- Addon: 1.2.0-36
```

> - Chart: 0.12.3

Reviewer 3: ```suggestion
- Chart: 0.12.16
```

> - kommander-federation (yakcl): 0.6.2

Reviewer 3: ```suggestion
- kommander-federation (yakcl): 0.6.9
```

> - kommander-licensing (yakcl): 0.6.2

Reviewer 3: ```suggestion
- kommander-licensing (yakcl): 0.6.9
```
